inspirehep/modules/migrator/models.py

```diff
@@ -52,3 +56,29 @@ def marcxml(self):
     @marcxml.setter
     def marcxml(self, value):
         self._marcxml = compress(value)
+
+    @hybrid_property
+    def error(self):
+        return self._errors
+
+    @error.setter
+    def error(self, value):
+        """errors column setter that stores an Exception and sets the ``valid`` flag."""
+        self.valid = False
+        self._errors = u'{}: {}'.format(type(value).__name__, value)
+
+    @classmethod
+    def from_marcxml(cls, raw_record):
+        """Create an instance from a MARCXML record.
+
+        The record must have a ``001`` tag containing the recid, otherwise it raises a ValueError.
+        """
+        try:
+            recid = int(cls.re_recid.search(raw_record).group('recid'))
+        except (ValueError, AttributeError):
+            raise ValueError('The MARCXML record contains no recid or recid is malformed')
```

Comment:
What about a more specific error such as ``MalfromedMARCXML`` or similar?  WDYT?

Reply:
I don't think it's very useful, as this is not something that should ever happen, it's more of a sanity check.